### ISSUE TITLE
Model cards for Keras models

### ISSUE THREAD

**Issue by merveenoyan** [huggingface/huggingface_hub#628]
Keras models (non-Transformers ones) don't have generated model cards.  I'd like to generate model cards for them that include:
- model hyperparameters
- metrics from training
- name of dataset and splits (can be filled by user later on as well) 
- versions of libraries (like TF model cards)

**Comment by merveenoyan** [comment]
I think it would be super cool to actually have [KerasTuner](https://keras.io/keras_tuner/) outputs in the model cards as well (output of `results_summary()`)

**Comment by merveenoyan** [comment]
I think it would be a big flex if we would write `model.summary()` output. This doesn't work for models with subclassing AFAIK, we can get it for Functional and Sequential models. This tends to be a large section but we can put it under toggle (pointed out by @osanseviero [in spaCy models](https://huggingface.co/spacy/fr_dep_news_trf))
The rest of the parameters I will get from model. 
We can later introduce history object (created with `history = model.fit()` that contains model training history even if the user doesn't log loss, accuracy etc, we could introduce this as another parameter in `push_to_hub_keras()` since this is something we can't obtain from the model but we need user to create history while fitting the model. 

I want to cc @Rocketknight1 since he's experienced with the TF model card.

**Comment by merveenoyan** [comment]
@Rocketknight1 it currently looks very shifted when I save it with IO so I'm plotting the model instead :( 
<img width="477" alt="Ekran Resmi 2022-02-08 13 14 02" src="https://user-images.githubusercontent.com/53175384/153030055-53533d16-cabe-4ae4-abec-d38ce25c922f.png">
